src/components/NewsletterSignup/NewsletterSignup.stories.tsx

```diff
@@ -0,0 +1,131 @@
+import type { Meta, StoryObj } from "@storybook/react";
+import { Box, VStack } from "@chakra-ui/react";
+import { withDesign } from "storybook-addon-designs";
+import NewsletterSignup from "./NewsletterSignup";
+import { sectionTypeArray } from "../../helpers/types";
+import Heading from "../Heading/Heading";
+
+const meta: Meta<typeof NewsletterSignup> = {
+  title: "Components/Form Elements/NewsletterSignup",
+  component: NewsletterSignup,
+  decorators: [withDesign],
+  parameters: {
+    jest: ["NewsletterSignup.test.tsx"],
+  },
+  argTypes: {
+    className: { control: false },
+    confirmationHeading: {
+      control: "text",
+    },
+    confirmationText: {
+      control: "text",
+    },
+    descriptionText: {
+      control: "text",
+      table: {
+        defaultValue: {
+          summary:
+            "Stay connected with the latest research news from NYPL, including information about our events, programs, " +
+            "exhibitions, and collections.",
+        },
+      },
+    },
+    formHelperText: {
+      control: "text",
+    },
+    id: { control: false },
+    newsletterSignupType: {
+      control: "select",
+      options: sectionTypeArray,
+      table: {
+        defaultValue: {
+          summary: "whatsOn",
+        },
+      },
+    },
+    onChange: { control: false },
+    onSubmit: { control: false },
+    title: {
+      control: "text",
+      table: { defaultValue: { summary: "Sign Up for Our Newsletter!" } },
+    },
+    valueEmail: { control: false },
+    view: {
+      control: "select",
+      options: ["form", "submitting", "confirmation", "error"],
+      table: {
+        defaultValue: {
+          summary: "form",
+        },
+      },
+    },
+  },
+};
+
+export default meta;
+type Story = StoryObj<typeof NewsletterSignup>;
+
+/**
+ * Main Story for the NewsletterSignup component. This must contains the `args`
+ * and `parameters` properties in this object.
+ */
+export const WithControls: Story = {
+  args: {
+    className: undefined,
+    confirmationHeading: "Thank you for signing up!",
+    confirmationText:
+      "You can update your email subscription preferences at any time using the links at the bottom of the email.",
+    descriptionText: undefined,
+    formHelperText: undefined,
+    id: undefined,
+    isInvalidEmail: false,
+    newsletterSignupType: undefined,
+    onChange: undefined,
+    onSubmit: undefined,
+    title: undefined,
```

Comment:
I noticed that when I try to update this prop in the controls, when it's empty the page breaks. I think it's because this should not be `undefined` by default.

Reply:
Good catch. But this is a tricky issue: I realized it has to to with the `Heading` component. If you go to the Heading Storybook entry it is having the same issue. I fixed it by adding a check in the Newsletter component itself even-though there is a default string always passed. I tried several things in the .stories.tsx file but could not get it to work. It might be possible that the fix needs to be added to the Heading component? I am not entirely sure. 🤔

Reply:
Yes, we should fix that in the `Heading` component but that's out of scope for this ticket. I'll create a follow up ticket for that.